tests/tools/utils.py

```diff
@@ -9,15 +9,20 @@
 from files import setdir
 
 
-def _chunk_numpy_array(arr, fmt_meta, chunk_sizes, path_fmt):
+def _chunk_numpy_array(arr, fmt_meta, chunk_sizes, path_fmt, vector_rows=False):
     paths = []
     offset = 0
 
     for j, n in enumerate(chunk_sizes):
         path = os.path.abspath(path_fmt % j)
         arr_chunk = arr[offset: offset + n]
+        shape = arr_chunk.shape
         logging.info("Chunking %d-%d" % (offset, offset + n))
-        array_readwriter.get_array_parser(**fmt_meta).write(path, arr_chunk)
+        # If requested we write multi-column arrays as single-column vector files
+        if vector_rows and len(shape) > 1 and shape[1] > 1:
```

Comment:
This check can be pushed down to the parquet writer, not sure what is preferred here. 

To guarantee that any Parquet file written using the parser can be read using the parser we might need to do that.

Reply:
writing a single-column is only available for `parquet`, right? we'd better add a format check here to capture any potential use `vector_rows=True` with `numpy`.

Reply:
It is actually possible with Numpy too, using a [structured array](https://numpy.org/doc/stable/user/basics.rec.html), but not sure if/why one would produce such a file if they already have a regular 2D `ndarray`.

My question is more about where these checks should exist, inside `ParquetArrayParser` or only at the tests?

Reply:
only at tests should be ok as `.write()` is called here for now.

Reply:
OK the new commit should make the writer only use vector_rows when we're writing to Parquet.